### PR TITLE
Add a way to import css in the files

### PR DESCRIPTION
**Pull request by SuelenKarbivnychyy**
Developed the logic to import CSS files and create a test file for it.
Updated README file with proper steps to successfully set up the environment to develop it locally.

### CODE REVIEW

**Comment by SuelenKarbivnychyy** [comment]
@sansyrox Please review.

File: starfyre/compiler.py [hunk 1]
```diff
@@ -2,13 +2,24 @@
...
     fyre_files = []
     for file in os.listdir(project_dir):
         if file.endswith(".fyre"):
             fyre_files.append(file)
     return fyre_files
 
+def import_css(import_statement):
+    """Read a css file and save it's content to a list"""
+    css_content = []    
+    css_file_name = import_statement.split("'")[1]
```
Comment (sansyrox) [inline starfyre/compiler.py:15]: This statement assumes that we only have import statements like `import 'test.css'`, but we would also want to support `import "test.css"`.


You can try a regular expression based approach here.

Something like this

```
import re

pattern = re.compile(r'import\s+["\'](.*?)\.css["\']')
match = pattern.search(statement1)
filename = match.group(1)
```

File: starfyre/compiler.py [hunk 2]
```diff
@@ -2,13 +2,24 @@
...
 
-def get_fyre_files(project_dir):
+def get_fyre_files(project_dir):   
     fyre_files = []
     for file in os.listdir(project_dir):
         if file.endswith(".fyre"):
             fyre_files.append(file)
     return fyre_files
 
+def import_css(import_statement):
```
Comment (sansyrox) [inline starfyre/compiler.py:12]: Also, one suggestion we can name it as `def resolve_css_import(import_statement):`

File: starfyre/compiler.py [hunk 3]
```diff
@@ -43,6 +54,10 @@ def remove_empty_lines_from_end(lines):
             elif line.startswith("--client"):
                 current_line_type = "client"  # this is a hack
                 continue
+            elif line.startswith("import") and ".css" in line:
+                css_content = import_css(line)
+                css_lines += css_content
+                continue
```
Comment (sansyrox) [inline starfyre/compiler.py:60]: nice!

File: starfyre/compiler.py [hunk 4]
```diff
@@ -146,11 +161,11 @@ def transpile_to_python(
 
     final_python_lines.append(main_content)
 
-    file_name = output_file_name.split("/")[-1]
+    file_name = output_file_name.split("/")[-1]                 #getting the file itself "without the path"
     output_file_name = project_dir / "build" / file_name
 
     with open(output_file_name, "w") as output_file:
-        output_file.write("".join(final_python_lines))
+        output_file.write("".join(final_python_lines))          #result of the transpiled
```
Comment (sansyrox) [inline starfyre/compiler.py:168]: Thank you for this!

File: README.md
Comment (sansyrox) [inline README.md:1]: Great work on the comments!

File: starfyre/compiler.py (hunk 1 above)
Comment (sansyrox) [reply starfyre/compiler.py:15]: Or maybe you can use the Path(https://docs.python.org/3/library/pathlib.html) function to resolve the path here. 

This would be the preferred approach but we can resort to regex if this doesn't work.

**Review by sansyrox — COMMENTED** [review]
Hey @SuelenKarbivnychyy 👋 

Great work on the PR! 😄  I love the additional docs!

I have a few inline comments. Do let me know if you require any more information :D

File: README.md [hunk 5]
```diff
@@ -94,17 +94,28 @@ If you still need help to get started, feel free to reach out on our community d
...
-1. Install the dependencies `poetry install`
-2. Run the script `./build.sh`. This command will run the build process in starfyre against the test application in `test-application` directory.
+Python version 3.10
+
+-Fork this repo:
+
+```
+https://github.com/sparckles/starfyre
+```
+
```
[the remark this answers is not in the document]
Comment (SuelenKarbivnychyy) [reply README.md]: Thank you. Suggestions applied :)

File: starfyre/compiler.py (hunk 2 above)
Comment (SuelenKarbivnychyy) [reply starfyre/compiler.py:12]: Thank you. Suggestion applied. :)

File: test-application/parent.fyre [hunk 6]
```diff
@@ -24,3 +24,5 @@ def ssr_request():
     </span>
 </pyml>
 
+import '/home/suelenk/starfyre_project/starfyre/test-application/css_file_test.css'
```
[the remark this answers is not in the document]
Comment (SuelenKarbivnychyy) [reply test-application/parent.fyre]: Thank you. I've corrected it. :)

File: starfyre/compiler.py (hunk 1 above)
Comment (SuelenKarbivnychyy) [reply starfyre/compiler.py]: Thank you for the suggestion. I implemented a simpler way to support both statements for now. Please take a look and let me know what you think. :)

File: README.md [hunk 7]
```diff
@@ -94,17 +94,25 @@ If you still need help to get started, feel free to reach out on our community d
 
 ## ⚙️  Developing Locally
 
-1. Install the dependencies `poetry install`
-2. Run the script `./build.sh`. This command will run the build process in starfyre against the test application in `test-application` directory.
+Python version 3.10
+
+1. Fork this [repo](https://github.com/sparckles/starfyre)
+2. Clone this repo - `git clone https://github.com/sparckles/starfyre`
+3. Go to in the starfire directory - `cd starfyre`
```
Comment (sansyrox) [inline README.md:101]: One nit - `in to` instead of `to in`

File: starfyre/compiler.py [hunk 8]
```diff
@@ -2,15 +2,30 @@
...
         if file.endswith(".fyre"):
             fyre_files.append(file)
     return fyre_files
 
+def resolve_css_import(import_statement, working_directory):
+    """Read a css file and save it's content to a list"""
+    css_content = []    
+    if "" in import_statement:
+        css_file_name = import_statement.replace('"', "'")        
+    css_file_name = css_file_name.split("'")[1]
```
Comment (sansyrox) [inline starfyre/compiler.py:17]: Hey @SuelenKarbivnychyy , I love the approach on line 16! Very smart!

However, line 17 still has a bug. This statement would still work on `import 'test.css`, which should not be the case. Using `split` will require us to handle a lot of edge cases.

File: starfyre/compiler.py [hunk 9]
```diff
@@ -2,15 +2,30 @@
...
     return fyre_files
 
+def resolve_css_import(import_statement, working_directory):
+    """Read a css file and save it's content to a list"""
+    css_content = []    
+    if "" in import_statement:
+        css_file_name = import_statement.replace('"', "'")        
+    css_file_name = css_file_name.split("'")[1]   
+    if css_file_name.startswith("."):
+        css_file_name = css_file_name.replace(".", str(working_directory), 1)
```
Comment (sansyrox) [inline starfyre/compiler.py:19]: If you use Path from pathlib(https://docs.python.org/3/library/pathlib.html) , the resolution will be implemeted by internal functions. It is always a better practice to use internal functions wherever possible.

**Review by sansyrox — COMMENTED** [review]
Hey @SuelenKarbivnychyy 👋 

Thank you for addressing the previous comments 😄  However, I have some more comments.

Do let me know if you require any more information from me 😄

File: starfyre/compiler.py (hunk 8 above)
Comment (SuelenKarbivnychyy) [reply starfyre/compiler.py]: Thank you again. I've addressed that out. :)

**Comment by sansyrox** [comment]
Hey @SuelenKarbivnychyy 👋 

Great work on the PR! There were a few minor suggestions, which I have addressed in a combined commit. The PR is almost ready for merging. 

Can you please rebase the latest changes from main to this branch?

**Comment by SuelenKarbivnychyy** [comment]
@sansyrox hey, Would you mind addressing the other suggestions here as well. I got lost on the combined commit you did.
I already got the latest version to this branch.

**Comment by sansyrox** [comment]
@SuelenKarbivnychyy , the final suggestion was merging the latest changes 😄
